### PR TITLE
For #101: Poster token retrieval

### PR DESCRIPTION
**Pull request by paulodamaso**
For #101:
- Added response for token retrieval in `poster_mock.py`
- Added puzzle for adding more test cases and refactoring `poster_mock.py`
- Renamed private methods on `Poster` (starting with `__`) to be used in `Poster` subclasses

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#145](https://github.com/timelesslounge/timelessis/issues/145) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#145](https://github.com/timelesslounge/timelessis/issues/145) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/b0a1981d-ad89-4dc3-a81c-dbf23d53eb80); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#145` already assigned to @vladarefiev, can't assign to @amihaiemil

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#145` already assigned to @vladarefiev, can't assign to @vladarefiev

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#145` already assigned to @vladarefiev, can't assign to @amihaiemil

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#145` already assigned to @vladarefiev, can't assign to @ammaratef45

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#145` already assigned to @vladarefiev, can't assign to @vladarefiev

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#145` already assigned to @vladarefiev, can't assign to @amihaiemil

File: timeless/poster/api.py [hunk 1]
```diff
@@ -23,40 +24,44 @@ def locations(self):
...
-        return self.token
+    def access_token(self):
+        """Fetches authorization tokens
+        
+        :return: Token retrieved from Poster api or error returned by poster pi
+        """
+        return self.send(
+            method=self.POST,
+            action="auth/access_token"
+        ).json()["access_token"]
```
Comment (vladarefiev) [inline timeless/poster/api.py]: @paulodamaso As I can see we don't handle cases when json hasn't `access_token`. here will be exception. first of all we should check response status code I think

File: tests/poster_mock.py [hunk 2]
```diff
@@ -40,7 +47,40 @@ def do_GET(self):
...
 
+        def do_POST(self):
+            if re.search(self.TOKEN_PATTERN, self.path):
+                self.send_response(requests.codes.ok)
+                self.send_header("Content-Type", "application/json; charset=utf-8")
+                self.end_headers()
+                content = json.dumps(
+                    {
+                        'access_token':'861052:02391570ff9af128e93c5a771055ba88',
+                        'account_number':'861052',
```
Comment (vladarefiev) [inline tests/poster_mock.py]: @paulodamaso Does it pass linter?
As I know we check double quotes before merging

File: tests/test_auth.py [hunk 3]
```diff
@@ -10,3 +19,32 @@ def test_forgot_password(client):
...
+        @todo #101:30min Implement other test cases for poster token 
+         retrieval. The remaining test cases must be discovered, the error 
+         codes in https://dev.joinposter.com/en/docs/api#authorization-in-api 
+         are not specific enough.   
+    """
+    def test_can_retrieve_token(self):
+        self.setup_class()
+        assert (
+            Authenticated(
+                url="http://localhost:{port}".format(port=self.port)
```
Comment (vladarefiev) [inline tests/test_auth.py]: @paulodamaso in python3.6 we just can do this `f"http://localhost:{self.port}"`

File: ittests/it_test_poster.py [hunk 4]
```diff
@@ -14,6 +14,4 @@ class PosterITTests(unittest.TestCase):
 
     @unittest.skip("poster.auth not implemented yet")
     def test_auth(self):
-        assert Authenticated(
-            Poster(clientid="$0m3C1i3ntId")
-        ).token() != "", "Poster did not authenticated user"
+        assert Authenticated(clientid="$0m3C1i3ntId").token() != "", "Poster did not authenticated user"
```
Comment (vladarefiev) [inline ittests/it_test_poster.py]: @paulodamaso I think we can do just `assert  not Authenticated(clientid="$0m3C1i3ntId").token(), "Poster did not authenticated user"`

**Comment by vladarefiev** [comment]
@paulodamaso good, but a few comments

**Comment by vladarefiev** [comment]
@paulodamaso good, but a few comments

**Comment by paulodamaso** [comment]
@vladarefiev Made the corrections, please take a look

File: timeless/poster/api.py [hunk 5]
```diff
@@ -50,10 +50,17 @@ def access_token(self):
...
+        response = self.send(
             method=self.POST,
             action="auth/access_token"
-        ).json()["access_token"]
+        )
+        if response.status_code != 200:
+            raise Exception("Problem acessing poster api")
+        token = response.json()["access_token"]
+        if (not token):
+            raise Exception("Token not found")
```
Comment (vladarefiev) [inline timeless/poster/api.py]: @paulodamaso let's do just
```
if not response.ok:
            raise Exception("Problem acessing poster api")
        token = response.json().get("access_token")
        if not token:
            raise Exception("Token not found")
```

**Comment by vladarefiev** [comment]
@paulodamaso one more comment

**Comment by paulodamaso** [comment]
@vladarefiev Done, please take a look

**Comment by vladarefiev** [comment]
@paulodamaso greatm thanks! fix conflicts and go to merge

**Comment by paulodamaso** [comment]
@vladarefiev Done, please take a look

**Comment by vladarefiev** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@paulodamaso there is an error, check rultor log

**Comment by paulodamaso** [comment]
@emilianodellacasa Sorry, didn't run the tests after the merge. Fixed problems but merge introduced some warnings (`DeprecationWarning: time.clock`) that are not related to my changes, please take a look.

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#145](https://github.com/timelesslounge/timelessis/issues/145) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**